astropy/utils/masked/core.py

```diff
@@ -1026,62 +1026,128 @@ def argmax(self, axis=None, out=None, *, keepdims=False):
+                    order = self.dtype.names
+                elif NUMPY_LT_2_0:
+                    order = np.core._internal._newnames(self.dtype, order)
+                else:
+                    order = np._core._internal._newnames(self.dtype, order)
 
-        Parameters
-        ----------
-        axis : int or None, optional
-            Axis along which to sort.  The default is -1 (the last axis).
-            If None, the flattened array is used.
-        kind : str or None, ignored.
-            The kind of sort.  Present only to allow subclasses to work.
-        order : str or list of str.
-            For an array with fields defined, the fields to compare first,
-            second, etc.  A single field can be specified as a string, and not
-            all fields need be specified, but unspecified fields will still be
-            used, in dtype order, to break ties.
+                keys = tuple(data[name] for name in order[::-1])
 
-        Returns
-        -------
-        index_array : ndarray, int
-            Array of indices that sorts along the specified ``axis``.  Use
-            ``np.take_along_axis(self, index_array, axis=axis)`` to obtain
-            the sorted array.
+            elif order is not None:
+                raise ValueError("Cannot specify order when the array has no fields.")
 
-        """
-        if axis is None:
-            data = self.ravel()
-            axis = -1
-        else:
-            data = self
-
-        if self.dtype.names:
-            # As done inside the argsort implementation in multiarray/methods.c.
-            if order is None:
-                order = self.dtype.names
-            elif NUMPY_LT_2_0:
-                order = np.core._internal._newnames(self.dtype, order)
             else:
-                order = np._core._internal._newnames(self.dtype, order)
+                keys = (data,)
+
+            return np.lexsort(keys, axis=axis)
+
+    else:
+
+        def argsort(self, axis=-1, kind=None, order=None, *, stable=False):
+            """Returns the indices that would sort an array.
+
+            Perform an indirect sort along the given axis on both the array
+            and the mask, with masked items being sorted to the end.
+
+            Parameters
+            ----------
+            axis : int or None, optional
+                Axis along which to sort.  The default is -1 (the last axis).
+                If None, the flattened array is used.
+            kind : str or None, ignored.
+                The kind of sort.  Present only to allow subclasses to work.
+            order : str or list of str.
+                For an array with fields defined, the fields to compare first,
+                second, etc.  A single field can be specified as a string, and not
+                all fields need be specified, but unspecified fields will still be
+                used, in dtype order, to break ties.
+            stable: bool, keyword-only
```

Comment:
Since it is not actually used, in this case I think we can get away with just defining `argsort` for all numpy and just not passing on `stable`.  Then, here write
```
Only for compatibility with np.sort (for numpy >= 2.0). Ignored.
```